tests/testlib/s2n_ktls_test_utils.c

```diff
@@ -138,10 +139,13 @@ ssize_t s2n_test_ktls_recvmsg_io_stuffer(void *io_context, struct msghdr *msg)
         if (no_more_records) {
             break;
         }
-        bool next_record_different_type = next_record_type != *record_type;
+
+        bool next_record_different_type = next_record_type != record_type;
         if (next_record_different_type) {
             break;
         }
+
+        POSIX_GUARD(s2n_stuffer_skip_read(&io_ctx->ancillary_buffer, sizeof(record_type)));
```

Comment:
Doesn't this change make the peek_char call kind of useless? Like, you could just read a uint8 instead of peeking and you wouldn't have to do this skip_read.

Reply:
Yeah I was wrong. I realize that you don't want to read the record type if you're not actually going to read the record because that would alter the stuffer and mess the code up when this function is called again.